src/async_nbgrader/async_nbgrader/apps/processmessageapp.py

```diff
@@ -0,0 +1,85 @@
+# coding: utf-8
+
+import base64
+import json
+import os
+import traceback
+
+from nbgrader.apps.baseapp import NbGrader
+from traitlets import default
+
+from ..helpers import chdir
+from ..helpers import get_nbgrader_api
+
+aliases = {
+    "log-level": "Application.log_level",
+    "db": "CourseDirectory.db_url",
+}
+flags = {}
+
+
+# JupyterHub settings
+JUPYTERHUB_API_URL = os.environ.get("JUPYTERHUB_API_URL") or "http://hub:8081/hub/api"
+JUPYTERHUB_BASE_URL = os.environ.get("JUPYTERHUB_BASE_URL") or "/"
+
+
+class ProcessMessageApp(NbGrader):
+    """App to handle amqp messages from Argo"""
+
+    name = "async_nbgrader-process-message"
+
+    aliases = aliases
+
+    @default("classes")
+    def _classes_default(self):
+        classes = super(ProcessMessageApp, self)._classes_default()
+        classes.append(ProcessMessageApp)
+        return classes
+
+    def start(self):
+        """Handler for processing mesage, message is passed as first argument, it is received in self.extra_args[0]"""
+        super(ProcessMessageApp, self).start()
+        if len(self.extra_args) == 0 or self.extra_args[0] == "":
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0` or `if len(val) &gt; 0`

 [Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)